### ISSUE TITLE
Polar contour plot / Contourpolar request

### ISSUE THREAD

**Issue by silvanbrummen** [plotly/plotly.js#4992]
Hi! 

Your plotting library is really awesome! Thanks for that!

I'm currently trying to make the switch from matplotlib (for Python plots) to Plotly (also Python) and I was looking for an option to display contour plots in a polar coordinate system. 

I came across the following issue in [plotly.py](https://github.com/plotly/plotly.py/issues/2024) where @emmanuelle suggested to create a post here in plotly.js. 

Do you guys have the intention of creating a Contourpolar type of chart? Similar to the plot show below (created in matplotlib):
![image](https://user-images.githubusercontent.com/46493049/87039727-e74ab980-c1ef-11ea-94a7-711ae5186e88.png)
([source]( https://github.com/plotly/plotly.py/issues/2024))

I've tried the Barpolar option, but it's not sufficient for the amount of detail I would like to visualize (a 360 degree water surface elevation plot), I think the plot I would like to display greatly benefits from being a contour plot.

Thanks in advance,
Sil

**Comment by nicolaskruchten** [comment]
This would be a great addition to the library, and we'd be happy to work with someone who wants to add it :)

**Comment by silvanbrummen** [comment]
Great! 

If there is any Python work that could be done in order to implement this kind of chart/trace, I'm willing to contribute. Although I do have the feeling that the Plotly codebase is JavaScript and all Python code is autogenerated from JavaScript. Is that correct?

**Comment by nicolaskruchten** [comment]
You're correct that the `plotly.graph_objects` Python module is largely auto-generated, but there is still a lot in Python e.g. `plotly.subplots`, `plotly.express` etc. New trace types need to be implemented in Javascript, but there is also the option of building a [Figure Factory](https://plotly.com/python/figure-factories/) which leverages existing trace types. For example we have a [ternary contour](https://plotly.com/python/ternary-contour/) figure factory which does something similar to what you're looking for: contours in non-cartesian coordinates. So if you wanted you could build a `create_polar_contour` figure factory along the same lines :)

**Comment by silvanbrummen** [comment]
Alright, good to know that there is an option to add to Python code directly. 

I'll give it a go by using the Figure Factory. Thanks for the help so far.

**Comment by archmoj** [comment]
FYI - Here is @emmanuelle's PR that added `ternary contour` to `plotly.py`
https://github.com/plotly/plotly.py/pull/1413

**Comment by coolbud140** [comment]
Hi there,

Thanks for all the wonderful work!

Has there been any development on this issue? Polar contour plot is something very useful for what I'm working on.

Thanks

**Comment by bernardopacini** [comment]
Hi @nicolaskruchten, I am not sure if there has been progress on this, but if not, I would be happy try to implement it if you can provide me some guidance on where to modify the JS and Python libraries.
